massa-execution-worker/src/execution.rs

```diff
@@ -70,6 +70,8 @@ pub(crate) struct ExecutionState {
     execution_interface: Box<dyn Interface>,
     // execution statistics
     stats_counter: ExecutionStatsCounter,
+    // cache of pre compiled sc modules
+    module_cache: Arc<RwLock<ModuleCache>>,
```

Comment:
When we talked about generics in the call the goal was to not have a dependency to wasmer and so have a generic module that wrap the wasmer module and maybe others in the future but don't have a type `ModuleCache`. I think the module cache should stay in Massa and use a `Module` defined in `massa-sc-runtime` this will avoid the `RwLock` and to pass this object to the execution.

Reply:
This would also be useful to have modules with differents VM/versions behind.

Reply:
> I think the module cache should stay in Massa and use a Module defined in massa-sc-runtime this will avoid the RwLock and to pass this object to the execution.

I was expecting this would cause a misunderstanding, that is not the thought process behind the lock, that is needed for reasons that have nothing to do with this issue (the runtime env needing to keep a reference to the cache in order to properly run subsequent modules). 

> When we talked about generics in the call the goal was to not have a dependency to wasmer and so have a generic module that wrap the wasmer module and maybe others in the future but don't have a type ModuleCache.

> This would also be useful to have modules with differents VM/versions behind.

As for the comments regarding `ModuleCache` being not generic enough I do not agree with it, what `ModuleCache` does can whatever we want, we just need to keep its information in execution between multiple runtime calls, the rest should not be handled in the node code.

Reply:
Updates will be made accordingly to what was decided with @AurelienFT & @sydhds, thought process behind the changes to be referenced here in the near future.

Reply:
This has been taken care of, resolving